templates/external/surveygizmo_daily_attitudes/import_responses.py

```diff
@@ -0,0 +1,118 @@
+  return dt.datetime.strftime(dt.date.fromisoformat(date_string) + dt.timedelta(days=1), '%Y-%m-%d')
+
+
+def format_response (s, date):
+    # Takes a single user's responses and returns a list of dictionaries, one per answer, formatted to corresponding bigquery fields.
+    # `survey_data` is a dict with question ID as the key and question/response details as the value
+    # e.g. survey_data: {'25': {'id': 25, 'type': 'RADIO', 'question': 'I trust Firefox to help me with my online privacy',
+    # 'section_id': 2, 'answer': 'Agree', 'answer_id': 10066, 'shown': True}}
+    resp = list(s.get("survey_data", {}).values())
+    try:
+        # Shield ID is sent as a hidden field with question name "Shield ID"
+        shield_id = [r.get("answer") for r in resp if r.get('question') == 'Shield ID'][0]
+    except IndexError:
+        shield_id = None
+
+    return [{
+        'shield_id': shield_id,
+        'date': date,
+        'question': r.get('question'),
+        'question_key': q_fields.get(r.get('question')),
+        'value': r.get('answer')
+    } for r in resp if r.get('question') != 'Shield ID']
+
+
+def construct_data(survey, date):
+    formatted = [format_responses(resp, date) for resp in survey['data']]
+    # flatten list of lists into a single list of dictionaries
+    return list(itertools.chain.from_iterable(formatted))
+
+
+def get_survey_data(survey_id, date_string, token, secret):
+    # per SurveyGizmo docs, times are assumed to be eastern
+    # https://apihelp.surveygizmo.com/help/filters-v5
+    # so UTC midnight must be converted to EST/EDT
+    start_date = utc_date_to_eastern_string(date_string)
+    end_date = utc_date_to_eastern_string(date_plus_one(date_string))
+
+    url = (f"https://restapi.surveygizmo.com/v5/survey/{survey_id}/surveyresponse"
+      f"?api_token={token}&api_token_secret={secret}&results_per_page=500"
+      # filter for date_submitted >= start_date
+      f"&filter[field][0]=date_submitted&filter[operator][0]=>=&filter[value][0]={start_date}"
+      # filter for date_submitted < end_date
+      f"&filter[field][1]=date_submitted&filter[operator][1]=<&filter[value][1]={end_date}")
+    resp = requests.get(url)
+    resp.raise_for_status()
+    survey = resp.json()
+
+    # if the result set is large, we'll have to page through them to get all data
+    total_pages = survey.get("total_pages")
+    print(f"Found {total_pages} pages after filtering on date={date_string}")
+
+    print("fetching page 1")
+    ret = construct_data(survey, date_string)
+
+    for page in range(2, total_pages+1):
+        print("fetching page {}".format(page))
+        resp = requests.get(url + f"&page={page}")
+        resp.raise_for_status()
+        ret = ret + construct_data(resp.json(), date_string)
+    return ret
+
+
+def insert_to_bq(data, table):
+    client = bigquery.Client()
+    print(f"Inserting {len(data)} rows into bigquery")
+    job = client.load_table_from_json(data, table)
```

Comment:
We should make this idempotent. To do so, we should specify the particular date partition as the destination, so something like `{table}${date.replace('-', '')}` to get something like `moz-fx-data-shared-prod.external.survey_gizmo_daily_attitudes$20191201`. And then we need to provide a LoadJobConfig that sets the write_disposition to overwrite.

You should be able to copy almost verbatim from https://github.com/mozilla/forecasting/blob/master/simpleprophet/simpleprophet/output.py#L97-L107 and https://github.com/mozilla/forecasting/blob/master/simpleprophet/simpleprophet/pipeline.py#L56-L57